internal/transport/http2_client.go

```diff
@@ -1661,7 +1678,11 @@ func (t *http2Client) keepalive() {
 				continue
 			}
 			if outstandingPing && timeoutLeft <= 0 {
-				t.Close(connectionErrorf(true, nil, "keepalive ping failed to receive ACK within timeout"))
+				err := connectionErrorf(true, nil, "keepalive ping failed to receive ACK within timeout")
+				if logger.V(logLevel) {
+					logger.Errorf("transport: closing transport due to error in keepalive handling: %v.", err)
```

Comment:
Did you actually look at how this log statement shows up in practice? Should you be avoiding some duplicate information?